text/0000-project-groups.md

```diff
@@ -0,0 +1,391 @@
+
+[rust-lang#54445]: https://github.com/rust-lang/rust/issues/54445
+
+# Guide-level explanation
+
+To address this confusion this RFC proposes breaking up the single "Working
+Group" term, into three distinct terms.
+
+- **Working Group** would map most directly to the previous "Domain Working
+  Group" terminology, focused on building and organising around new areas of
+  Rust that aren't covered by the current team roster.
+- **Project Group** would replace the "Team Working Group", and would serve as a
+  catch all term for sub teams for the existing teams to organise around
+  specific efforts, such as certain project or effort in the Rust team.
+- **Community Group** would act as a catch all term for community self
+  organising groups that are independent of the Rust Programming
+  Language Organisation.
+
+## Life-cycle of a Project Group
+
+This is a high level overview of the complete process of a project group. While
+the flow is built around project groups, we expect that working groups would
+follow a similar process with only minor specifics changed. E.g. A working
+group does not have to find a liaison.
+
+<p align="center">
+    <img src="../resources/project-group-workflow.png"
+        alt="A flow diagram showing each step of a project group"
+        height="800px">
+    <p align="center">Figure 1. Project Group Lifecycle</p>
+</p>
+
+### Steps
+
+1. Exploratory period.
+
+- Initial discussions of the problem area.
+- Write a charter containing motivation, and some notes on
+  possible solutions.
+- Find a person from the relevant team who's willing to act as a liaison.
+  - Typically can find someone by creating a post on [internals] or pinging
+    specific people from team to gauge their interest.
+
+2. Obtain consensus to create group.
+
+- Specify the liaison, and shepherd(s).
+- How consensus is reached would vary from team to team, some would require an
+  RFC while others could decide in a meeting. (See [Future Work](#future-work))
+
+3. Create infrastructure for group.
+
+- GitHub repository under `rust-lang` for hosting work and discussions, such
+  as for draft RFCs.
+- A Discord channel or a Zulip stream for communication.
+- Project group in [`rust-lang/team`], as well as a team on GitHub, for
+  handling permissions.
+
+4. Create a post on the Rust or Inside Rust blog announcing creation of
+   the group.
+
+5. The group works towards the goals laid out in their charter.
+
+6. When active work has stopped a group is "archived".
+
+- Archival is not necessarily a permanent state, it is only a reflection on the current
+  status of the group. A group can be "restored" at a later stage.
+- Reasons to archive:
+  - Nobody in the group has time anymore or higher priority things arose.
+  - There's a blocking issue that can't be resolved.
+  - Don't see any additional work to do in this area in the near future.
+  - The work was done to a satisfactory state.
+  - The group decided the idea wasn't so good after all.
```

Comment:
We initially had distinct states, but then collapsed them down to one. I think the idea was "it doesn't make much difference at the end of the day", i.e., we need to include a write-up of what the problems were, and if we have that, there is always the option of e.g. renewing an effort (via RFC, as you note), so what does it matter if they "ran out of steam" or "got blocked" or "decided the idea wasn't so great".

But I'm not opposed to having either more states or perhaps just a "reason" that is selected from a narrow list of options (perhaps with a "supply your own" possibility). If nothing else, it'd be useful for a 

What do you think the set of completion states are? The set you listed doesn't seem complete to me (it feels like we should have something like "closed because we decided it wasn't such a great idea after all"), if nothing else.

Reply:
@nikomatsakis 
> But I'm not opposed to having either more states or perhaps just a "reason" that is selected from a narrow list of options (perhaps with a "supply your own" possibility). If nothing else, it'd be useful for a

You accidentally missed writing the rest of this sentence...

Reply:
I've updated the text to say non-exhaustive, as I don't feel like we should have a explicit set of states and these should really been seen as examples. Because to me any reason is valid, and exactly how a group how finishes isn't the important aspect to me compared with properly preserving their work so that it can act as a future reference.